src/main/java/org/opentripplanner/middleware/otp/OtpDispatcherImpl.java

```diff
@@ -0,0 +1,81 @@
+package org.opentripplanner.middleware.otp;
+
+import org.opentripplanner.middleware.otp.response.Response;
+import org.opentripplanner.middleware.utils.JsonUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.ws.rs.core.UriBuilder;
+import java.io.IOException;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.time.Duration;
+
+/**
+ * Responsible for constructing requests to elected OTP server end point using original query parameters provided by MOD UI.
+ * To provide the response from the OTP server in the form of status code and body so the correct response can be provided to the calling MOD UI.
+ */
+public class OtpDispatcherImpl implements OtpDispatcher {
```

Comment:
I'm not a Java expert, so I don't quite understand the point of an Interface if there is just one implementation. Is there a good article I can read up on why this might be a good practice?

Reply:
There probably isn't much point if there is only one implementation. I always (rightly or wrongly) code to interfaces. If later on down the line there is a second implementation the impact will be minimal if any. Of course you already know this! Articles focus on interface best practises and not on single implementation. If you feel an interface isn't warranted, I can remove it.

@landonreed @binh-dam-ibigroup Thoughts?